### PR TITLE
Validate DorogayaSet input bounds

### PR DESCRIPTION
**Pull request by krotname**
### Motivation
- Prevent crashes and heap exhaustion when the CLI is fed malicious or malformed stdin by validating graph sizes and vertex IDs before allocating arrays or calling DSU operations.

### Description
- Add bounded limits `MAX_VERTICES` and `MAX_EDGES` and helper methods `isValidGraph` and `isValidVertex` to centralize validation.
- Check graph validity at the start of `solve(int, Edge[])` and validate `n` and `m` in `run()` before allocating `Edge[]`, returning the existing failure result for invalid input.
- While reading edges, check each endpoint and avoid invoking `solve` when any endpoint is out of range to prevent `ArrayIndexOutOfBoundsException` or large allocations.
- Add two CLI regression test cases in `AlgorithmCliTest` covering a negative edge count and an out-of-range endpoint that both expect the existing failure message.

### Testing
- Ran the full test suite with `mvn test` and the build succeeded with all tests passing (`Tests run: 548, Failures: 0, Errors: 0, Skipped: 0`).
- The updated `algorithms.AlgorithmCliTest` and `algorithms.EmbeddedAlgorithmSelfChecksTest` were exercised as part of the run and passed.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a3cbf2fea7883278dbe74e7d441298d)

### CODE REVIEW

File: src/main/java/algorithms/sprint6/DorogayaSet.java [hunk 1]
```diff
@@ -236,17 +260,28 @@ private static void run() throws Exception {
...
         int m = in.nextInt();
 
-        Edge[] edges = new Edge[m];
+        long answer = -1;
 
-        for (int i = 0; i < m; i++) {
-            int from = in.nextInt();
-            int to = in.nextInt();
-            int weight = in.nextInt();
+        if (n >= 1 && n <= MAX_VERTICES && m >= 0 && m <= MAX_EDGES) {
```
[the remark this answers is not in the document]
Comment (krotname) [reply src/main/java/algorithms/sprint6/DorogayaSet.java:265]: Fixed in consolidated #116 (0e23996): FastIn now uses checked negative accumulation, rejects values outside int range before graph bounds, and accepts both exact int endpoints.

**Comment by krotname** [comment]
Закрываю как superseded by #116: изменения перенесены в одну проверенную ветку, все review findings учтены; локальный mvn verify прошёл полностью.
